src/coordinator/functions/abs_test.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package functions
+
+import (
+	"math"
+	"testing"
+
+	"github.com/m3db/m3db/src/coordinator/parser"
+	"github.com/m3db/m3db/src/coordinator/test"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func expectedValues(values [][]float64) [][]float64 {
+	expected := make([][]float64, 0, len(values))
+	for _, val := range values {
+		v := make([]float64, len(val))
+		for i, ev := range val {
+			v[i] = math.Abs(ev)
+		}
+
+		expected = append(expected, v)
+	}
+	return expected
+}
+
+func TestAbsWithAllValues(t *testing.T) {
+	values, bounds := test.GenerateValuesAndBounds(nil, nil)
+	values[0][0] = -values[0][0]
+	values[1][1] = -values[1][1]
+
+	block := test.NewBlockFromValues(bounds, values)
+	c, sink := test.NewControllerWithSink(parser.NodeID(1))
+	node := (&AbsOp{}).Node(c)
+	err := node.Process(parser.NodeID(0), block)
+	require.NoError(t, err)
+	expected := expectedValues(values)
+	assert.Len(t, sink.Values, 2)
+	assert.Equal(t, expected, sink.Values)
+}
+
+func TestAbsWithSomeValues(t *testing.T) {
```

Comment:
Worth it to write a test with all `NaN`s?

Reply:
Functionally one Nan is similar to all NaNs since there's no special logic there.